google/cloud/storage/examples/storage_bucket_default_kms_key_samples.cc

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/client.h"
+#include "google/cloud/storage/examples/storage_examples_common.h"
+#include "google/cloud/internal/getenv.h"
+#include <functional>
+#include <iostream>
+#include <map>
+
+namespace {
+
+using google::cloud::storage::examples::Commands;
+using google::cloud::storage::examples::CommandType;
+using google::cloud::storage::examples::Usage;
+
+void AddBucketDefaultKmsKey(google::cloud::storage::Client client,
+                            std::vector<std::string> const& argv) {
+  //! [add bucket kms key] [START storage_set_bucket_default_kms_key]
+  namespace gcs = google::cloud::storage;
+  using ::google::cloud::StatusOr;
+  [](gcs::Client client, std::string bucket_name, std::string key_name) {
+    StatusOr<gcs::BucketMetadata> updated = client.PatchBucket(
+        bucket_name, gcs::BucketMetadataPatchBuilder().SetEncryption(
+                         gcs::BucketEncryption{key_name}));
+    if (!updated) throw std::runtime_error(updated.status().message());
+
+    if (!updated->has_encryption()) {
+      std::cerr << "The change to set the encryption attribute on bucket "
+                << updated->name()
+                << " was sucessful, but the encryption is not set."
+                << "This is unexpected, maybe a concurrent change?\n";
+      return;
+    }
+
+    std::cout << "Successfully set default KMS key on bucket  "
```

Comment:
Nit: looks like two spaces in this string. Not sure if that's on purpose.

Reply:
Thanks, fixed.